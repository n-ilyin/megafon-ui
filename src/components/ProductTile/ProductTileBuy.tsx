import * as React from 'react';
import { cnCreate } from '../../utils/cn';
import './style/ProductTileBuy.less';
import Button from '../Button/Button';
import TextLink from '../TextLink/TextLink';
import { IProductTileBuy } from './types';
import { productTileBuyProps } from './propTypes';

const cn = cnCreate('mfui-product-tile-buy');
class ProductTileBuy extends React.Component<IProductTileBuy> {
    static propTypes = productTileBuyProps;

    render() {
        const {
            className,
            buyButton = {},
            connectButton = {},
            onClickBuy,
            onClickConnect,
        } = this.props;
        const {
            buyLink,
            buyButtonText,
            buttonPassiveColor,
            showBuyButton,
        } = buyButton;
        const {
            connectLink,
            connectButtonText,
            showConnectButton,
        } = connectButton;

        return (
            <div className={cn('', {}, className)}>
                {showBuyButton &&
                    <Button
                        className={cn('button', { 'single-button': !showConnectButton })}
                        passiveColor={buttonPassiveColor}
                        hoverColor="green"
                        sizeAll="medium"
                        href={buyLink}
                        onClick={onClickBuy}
                    >
                        {buyButtonText}
                    </Button>
                }
                {showConnectButton &&
                    <TextLink
                        className={cn('detail-link')}
                        href={connectLink}
                        target="_blank"
                        onClick={onClickConnect}
                    >
                        {connectButtonText}
                    </TextLink>
                }
            </div>
        );
    }
}

export default ProductTileBuy;
