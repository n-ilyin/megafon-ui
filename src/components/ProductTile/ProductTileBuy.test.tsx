import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTileBuy from './ProductTileBuy';

const props = {
    className: 'class',
    buyButton: {
        buyLink: '#',
        buyButtonText: 'Купить',
        showBuyButton: true,
        buttonPassiveColor: 'green' as 'green',
    },
};

describe('<ProductTileBuy />', () => {
    it('irenders empty ProductTileBuy', () => {
        const wrapper = shallow(<ProductTileBuy />);

        expect(wrapper).toMatchSnapshot();
    });

    it('irenders ProductTileBuy', () => {
        const wrapper = shallow(<ProductTileBuy {...props} onClickBuy={jest.fn()} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders ProductTileBuy with connectButton', () => {
        const connectButton = {
            connectLink: '#',
            connectButtonText: 'Подключить',
            showConnectButton: true,
        };
        const wrapper = shallow(<ProductTileBuy {...props} connectButton={connectButton} onClickConnect={jest.fn()}/>);

        expect(wrapper).toMatchSnapshot();
    });
});
