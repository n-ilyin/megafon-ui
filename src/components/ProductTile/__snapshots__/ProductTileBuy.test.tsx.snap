// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTileBuy /> irenders ProductTileBuy 1`] = `
<div
  className="mfui-product-tile-buy class"
>
  <Button
    className="mfui-product-tile-buy__button mfui-product-tile-buy__button_single-button"
    disabledColor="gray"
    downColor="dark"
    hoverColor="green"
    href="#"
    onClick={[MockFunction]}
    padding={true}
    passiveColor="green"
    showSpinner={false}
    sizeAll="medium"
    type="button"
    width="auto"
  >
    Купить
  </Button>
</div>
`;

exports[`<ProductTileBuy /> irenders empty ProductTileBuy 1`] = `
<div
  className="mfui-product-tile-buy"
/>
`;

exports[`<ProductTileBuy /> renders ProductTileBuy with connectButton 1`] = `
<div
  className="mfui-product-tile-buy class"
>
  <Button
    className="mfui-product-tile-buy__button"
    disabledColor="gray"
    downColor="dark"
    hoverColor="green"
    href="#"
    padding={true}
    passiveColor="green"
    showSpinner={false}
    sizeAll="medium"
    type="button"
    width="auto"
  >
    Купить
  </Button>
  <TextLink
    className="mfui-product-tile-buy__detail-link"
    color="blue"
    href="#"
    onClick={[MockFunction]}
    target="_blank"
    underlineStyle="solid"
    underlineVisibility="hover"
  >
    Подключить
  </TextLink>
</div>
`;
